File: baekjoon/단계별로 풀어보기/스택/10828_스택.py
# https://www.acmicpc.net/problem/10828

from sys import stdin

n = int(input())
stack = []
for _ in range(n):
    s = stdin.readline()
    try:
        if s.startswith('push'):
            stack.append(int(s.split()[-1]))
        elif s.startswith('pop'):
            print(stack.pop(-1))
        elif s.startswith('size'):
            print(len(stack))
        elif s.startswith('empty'):
            print(0) if stack else print(1)
        elif s.startswith('top'):
            print(stack[-1])
    except:
        print(-1)

File: C#/Baekjoon/스택/10828_스택.cs
//https://www.acmicpc.net/problem/10828

using var sr = new StreamReader(new BufferedStream(Console.OpenStandardInput()));
using var sw = new StreamWriter(new BufferedStream(Console.OpenStandardOutput()));
Stack<int> s = new Stack<int>();
string l;
int n = int.Parse(sr.ReadLine()), i;
while (n-- > 0)
{
    l = sr.ReadLine();
    if (l == "pop") sw.WriteLine(s.TryPop(out i) ? i : -1);
    else if (l == "size") sw.WriteLine(s.Count);
    else if (l == "empty") sw.WriteLine(s.Count == 0 ? 1 : 0);
    else if (l == "top") sw.WriteLine(s.Count == 0 ? -1 : s.Peek());
    else s.Push(int.Parse(l.Split()[1]));
}